events/janus_rabbitmqevh.c

```diff
@@ -586,3 +616,40 @@ static void *janus_rabbitmqevh_handler(void *data) {
 	JANUS_LOG(LOG_VERB, "Leaving RabbitMQEventHandler handler thread\n");
 	return NULL;
 }
+
+
+/* Thread to handle heartbeats */
+static void *janus_rabbitmqevh_heartbeat_handler(void *data) {
+	JANUS_LOG(LOG_VERB, "Monitoring RabbitMQ HeartBeat\n");
+		struct timeval timeout;
+	timeout.tv_sec = 0;
+	timeout.tv_usec = 20000;
```

Comment:
Shouldn't this be set to whatever the heartbeat was configured to?

Reply:
Initially, I thought that timeout.tv values should be shorter to identify connection issues with RabbitMQ immediately. However, after more tests, were noticed that even with larger timeout.tv values defined when an error occurs the module tries to reconnect.
Since the heartbeat frames are sent about half of the time defined on the heartbeat I decided to change the value to a bit more than half of the heartbeat.